pandas/core/ops/array_ops.py

```diff
@@ -158,15 +159,18 @@ def na_arithmetic_op(left, right, op, str_rep: Optional[str], is_cmp: bool = Fal
     """
     import pandas.core.computation.expressions as expressions
 
-    try:
-        result = expressions.evaluate(op, str_rep, left, right)
-    except TypeError:
-        if is_cmp:
-            # numexpr failed on comparison op, e.g. ndarray[float] > datetime
-            #  In this case we do not fall back to the masked op, as that
-            #  will handle complex numbers incorrectly, see GH#32047
-            raise
-        result = masked_arith_op(left, right, op)
+    with warnings.catch_warnings():
+        # suppress warnings from numpy about element-wise comparison
+        warnings.simplefilter("ignore", DeprecationWarning)
```

Comment:
What warning are we suppressing here, and is this future-proof? If this is the usual

```python
In [2]: np.array([1, 2]) == 'a'
/Users/taugspurger/.virtualenvs/dask-dev/bin/ipython:1: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  #!/Users/taugspurger/Envs/dask-dev/bin/python
Out[2]: False
```

then in the future the result will be `array([False, False])`. Is that what we want?

Reply:
this warning-catching was necessary to make the npdev build, pass, im going to see if i can revert it

Reply:
looks like catching this is needed in some form, but i can move the catching so as to cast a less-wide net.